core/src/main/scala/codesearch/core/util/Helper.scala

```diff
@@ -38,33 +38,34 @@ object Helper {
   def readFileAsync(path: String): IO[List[String]] =
     IO(Source.fromFile(path, "UTF-8")).bracket(source => IO.pure(source.getLines.toList))(source => IO(source.close()))
 
-  def hideSymbols(str: String): String = {
-    str.foldRight("") {
-      case (c, res) if SPECIAL_CHARS contains c =>
-        s"\\$c$res"
-      case (c, res) =>
-        s"$c$res"
+  def preciseMatch(query: String): String = {
+    val queryTokens: Seq[Token]   = Tokenizer.parseStringWithSpecialSymbols(query)
+    val specialChars: Set[String] = Set("|", "^", "$", "$", "+", "*", "(", ")", ".", "?")
```

Comment:
Later we should use the tokenizer to do this, but for now this is fine.

Reply:
That is, use tokens instead of `Set`?

Reply:
Look at https://github.com/aelve/codesearch/pull/172/commits/3b010c5032e42df7f0bef7a5f689ef9e88aa3067. Exactly what is needed?

Reply:
We should basically always escape SpecialSymbols and not have this `specialChars` thing at all -- this is the end goal